File: camerax_android/android/src/main/kotlin/dev/hebei/camerax_android/video/FallbackStrategyImpl.kt
package dev.hebei.camerax_android.video

import androidx.camera.video.FallbackStrategy
import androidx.camera.video.Quality
import dev.hebei.camerax_android.CameraXImpl
import dev.hebei.camerax_android.PigeonApiFallbackStrategyApi

class FallbackStrategyImpl(impl: CameraXImpl) : PigeonApiFallbackStrategyApi(impl) {
    override fun higherQualityOrLowerThan(quality: Quality): FallbackStrategy {
        return FallbackStrategy.higherQualityOrLowerThan(quality)
    }

    override fun higherQualityThan(quality: Quality): FallbackStrategy {
        return FallbackStrategy.higherQualityThan(quality)
    }

    override fun lowerQualityOrHigherThan(quality: Quality): FallbackStrategy {
        return FallbackStrategy.lowerQualityOrHigherThan(quality)
    }

    override fun lowerQualityThan(quality: Quality): FallbackStrategy {
        return FallbackStrategy.lowerQualityThan(quality)
    }
}

File: camerax_ios/lib/src/video/fallback_strategy_impl.dart
// import 'package:camerax_ios/src/api.dart';
// import 'package:camerax_ios/src/video.dart';
// import 'package:camerax_platform_interface/camerax_platform_interface.dart';

// final class FallbackStrategyImpl implements FallbackStrategy {
//   final FallbackStrategyProxyApi api;

//   FallbackStrategyImpl.internal(this.api);
// }

// final class FallbackStrategyChannelImpl extends FallbackStrategyChannel {
//   @override
//   FallbackStrategy higherQualityOrLowerThan(Quality quality) {
//     final api = FallbackStrategyProxyApi.higherQualityOrLowerThan(
//       quality: quality.api,
//     );
//     return FallbackStrategyImpl.internal(api);
//   }

//   @override
//   FallbackStrategy higherQualityThan(Quality quality) {
//     final api = FallbackStrategyProxyApi.higherQualityThan(
//       quality: quality.api,
//     );
//     return FallbackStrategyImpl.internal(api);
//   }

//   @override
//   FallbackStrategy lowerQualityOrHigherThan(Quality quality) {
//     final api = FallbackStrategyProxyApi.lowerQualityOrHigherThan(
//       quality: quality.api,
//     );
//     return FallbackStrategyImpl.internal(api);
//   }

//   @override
//   FallbackStrategy lowerQualityThan(Quality quality) {
//     final api = FallbackStrategyProxyApi.lowerQualityThan(quality: quality.api);
//     return FallbackStrategyImpl.internal(api);
//   }
// }

// extension FallbackStrategyX on FallbackStrategy {
//   FallbackStrategyProxyApi get api {
//     final impl = this;
//     if (impl is! FallbackStrategyImpl) throw TypeError();
//     return impl.api;
//   }
// }
